functions/go/apply-replacements/run_js.go

```diff
@@ -37,19 +41,40 @@ func applyReplacements(input []byte) ([]byte, error) {
 	return fn.Run(fn.ResourceListProcessorFunc(replacements.ApplyReplacements), []byte(input))
 }
 
-func resourceListProcessorWrapper() js.Func {
-	// TODO: figure out a better way to surface a golang error to JS environment.
-	// Currently error is surfaced as a string.
-	jsonFunc := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+func resourceListProcessorWrapper(resourceList *[]byte) js.Func {
+	jsonFunc := js.FuncOf(func(this js.Value, args []js.Value) any {
 		if len(args) != 1 {
 			return "Invalid number of arguments passed"
 		}
 		input := args[0].String()
 		applied, err := applyReplacements([]byte(input))
 		if err != nil {
-			return fmt.Errorf("unable to process resource list:", err.Error())
+			*resourceList = applied
+			return "unable to process resource list: " + err.Error()
 		}
+		*resourceList = applied
 		return string(applied)
 	})
+
 	return jsonFunc
 }
+
+func resourceListProcessorErrors(resourceList *[]byte) js.Func {
+	jsonFunc := js.FuncOf(func(this js.Value, args []js.Value) any {
+		rl, err := fn.ParseResourceList(*resourceList)
+		if err != nil {
+			return ""
+		}
+		if len(rl.Results) == 0 {
+			return ""
+		}
+		errorMessages := ""
+		for _, r := range(rl.Results) {
```

Comment:
When running in a pipeline, the function only appends `Results` to `ResourceList.Results` but not override the existing ones. So this  `resourceListProcessorErrors ` may get irrelevant error `Results` (from previous functions' results). 

I'd suggest at least adding a comment to notify future contributors about this since it could cause some unexpected failures in certain risk conditions.

Reply:
Added!